src/__test__/containers/signup.test.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import mockAxios from 'axios';
+import { Signup } from '../../Containers/Signup/Signup';
+import { signup } from '../../actions/signupActions';
+import signupReducer from '../../reducers/signupReducer';
+import { SIGNUP_ERROR, SIGNUP_SUCCESS, SIGNUP_STARTED } from '../../actionTypes/signupTypes';
+
+/**
+ * Signup handler
+ *
+ * @returns {function} then
+ */
+const onSignup = () => {
+  return Promise.resolve({ status: 201 });
+};
+
+const props = {
+  error: 'none',
+  onSignup,
+  history: {
+    /**
+     *Pushes a route
+     *
+     * @returns {object} mock
+     */
+    push() {
+      return jest.fn('');
+    },
+  },
+};
+
+const user = {
+  username: 'username',
+  email: 'email@email.com',
+  password: 'password',
+};
+
+const mockStore = configureMockStore([thunk]);
+
+describe('Render Signup component', () => {
+  it('to have wrapper class', async () => {
+    const wrapper = shallow(<Signup {...props} />);
+    expect(wrapper.find('.wrapper').length).toBe(1);
+  });
+
+  it('to should signup a user', async () => {
+    const wrapper = mount(<Signup {...props} />);
+    wrapper.find('input[name="username"]').simulate('input', { target: { value: 'username' } });
+    wrapper.find('input[name="email"]').simulate('input', { target: { value: 'grace@gmail.com' } });
+    wrapper.find('input[name="password"]').simulate('input', { target: { value: 'Abc123' } });
+    wrapper
+      .find('input[name="confirmPassword"]')
+      .simulate('input', { target: { value: 'Abc123' } });
+    wrapper.find('form').simulate('submit');
+    expect(wrapper.instance().props.history.push).toEqual(expect.any(Function));
+  });
+
+  it('to should return an error when passwords do not match', async () => {
+    const wrapper = mount(<Signup {...props} />);
+    wrapper.find('input[name="username"]').simulate('input', { target: { value: 'username' } });
+    wrapper.find('input[name="email"]').simulate('input', { target: { value: 'grace@gmail.com' } });
+    wrapper.find('input[name="password"]').simulate('input', { target: { value: 'Abc123' } });
+    wrapper
+      .find('input[name="confirmPassword"]')
+      .simulate('input', { target: { value: 'no match' } });
+    wrapper.find('form').simulate('submit');
+    expect(wrapper.instance().state.noMatch).toBe('Passwords do not match');
+  });
+});
+
+describe('Signup actions', () => {
+  let store;
+  beforeEach(() => {
+    store = mockStore({
+      events: {},
```

Comment:
@gracelungu, I think this `events` is not supposed to be here. Can you change it into something like user?